### DISCUSSION TITLE
Is there a built-in method for getting the current state?

### DISCUSSION THREAD

**Discussion by ari-sc** [andrew-gresyk/HFSM2#95] [Q&A] [answered]
While it's possible to get the current state using the "isActive" method and checking all the possible states, a "getActiveState" would be really nice to have.

**Comment by andrew-gresyk** [comment, marked as answer]
That's easy to do in a flat FSM, e.g. [FFSM2](https://flat.hfsm.dev) provides [StateID Root::activeStateId()](https://github.com/andrew-gresyk/FFSM2/blob/6ac88060b277c95c5dc31338caab188d02ae3ffe/development/ffsm2/detail/root_0.hpp#L137).

However, a hierarchical FSM with depth of 2 or more will have more than one active state.

**Reply by ari-sc** [reply]
Got it. Thanks for the response!
